src/main/java/org/tugraz/sysds/runtime/controlprogram/context/ExecutionContext.java

```diff
@@ -367,7 +373,11 @@ public MatrixObject allocateGPUMatrixObject(String varName, long numRows, long n
 			e = e1;
 		}
 		try {
-			dim2 = validateDimensions(mo.getNumColumns(), numCols);
+			// ToDo: fix this hack
+			if(instName.equals("gpu_ucumk+*"))
+				dim2 = validateDimensions(1, numCols);
+			else
+				dim2 = validateDimensions(mo.getNumColumns(), numCols);
 		} catch(DMLRuntimeException e1) {
 			e = e1;
 		}
```

Comment:
Since you're on it, please fix this hack of dimension validation via exception handling.